### ISSUE TITLE
Update for Hero4

### ISSUE THREAD

**Issue by hampld** [r1pper/GoPro.Hero#14]
Hi, 

is there any chance of getting an update for the hero4? 

cheers

**Comment by r1pper** [comment]
Hi,
I would really love to, but unfortunately at the moment I don't have access to a Hero 4 camera :(

Best regards

**Comment by hampld** [comment]
Sorry for the late reply. Would you be willing to update the lib, should you gain access to a Hero4?

**Comment by r1pper** [comment]
Sorry, Its been a busy week at work, Yes I definitely want to update the library but I need to have access to a Hero4 (for start, I at least need its sniffed files (using wireshark or similar tools) and available commands (locate, mode,...) based on Gopro application (android or iOS) and someone with a camera who is willing to test it)

Best regards

**Comment by hampld** [comment]
No problem! As I need the lib quite urgently (and currently don't have the time to do it myself...), I'd be willing to donate a Hero4 session towards an update of the lib. Is this something you're interested in?

**Comment by r1pper** [comment]
Oh, it’s very generous of you, yes I’m very interested, the problem is I live in Iran and I’m not sure if it’s plausible for you to do it.

By the way if you are not bounded to .Net there is a library for [Hero4](https://github.com/DenisCarriere/gopro) in python which is written by [Denis Carriere](https://github.com/DenisCarriere).
But if you are interested in .Net library, and deduced that you suggestion is not feasible, you can send me sniffed pcap files of your camera and I can work on the update with it this weekend and see how it goes. 

And could you please tell me how much time you have?

Best regards
Hessam

**Comment by hampld** [comment]
Well, yes ... that could be a small problem. I assume you don't get amazon deliveries in Iran. I'll check with my Iranian contacts whether we can work around this problem. 

Unfortunately my whole project is .NET, so writing a wrapper for a python lib is not my preferred option. I'm out of the office most of this week, but next week I could create a few pcap files (however, you'll need to tell me specifically what you need). The time frame is around 3 to 4 weeks. So it's not super urgent, but also not end-of-year. 

Best, 
David

**Comment by r1pper** [comment]
Yes, You are right, amazon does not have shipping services for Iran, but we have our own amazon like store [Digikala](www.digikala.com) which has some GoPro cameras available now.

I understand, I'm waiting for your contact whenever you have time and have access to the camera, then we can get through it together and generate required files.

Best regards
Hessam

**Comment by armandvanderwalt** [comment]
Hi,

Any news on GoPro 4 integration?

Thanks

**Comment by r1pper** [comment]
@armandvanderwalt  Hi,

I'm really interested in adding Hero4 to the library, but unfortunately I don't own the camera, hence I need help for testing and packet sniffing (albeit not necessary for initial support, we can use information from other libraries)

so please let me know if you are interested in collaborating Hero 4 integration.

Best regards

**Comment by armandvanderwalt** [comment]
I will be getting my hands on a GOPRO 4 soon then I can assist in this

**Comment by r1pper** [comment]
@armandvanderwalt Great, please let me know when you are ready, meanwhile I try to gather some information and do the initial implementation in a separate branch.

**Comment by KonradIT** [comment]
@r1pper get all the HERO4 URLs here: http://github.com/konradit/goprowifihack

https://github.com/KonradIT/goprowifihack/blob/master/HERO4/WifiCommands.md

**Comment by r1pper** [comment]
@KonradIT thanks, hopefully I'll start the development this weekend.

**Comment by KonradIT** [comment]
Ask me all you need this weekend before I go offline (I have to study :disappointed:  )
